File: languages/php/Bloock/Entity/Identity/CredentialOfferBody.php
<?php

namespace Bloock\Entity\Identity;

use Google\Protobuf\Internal\RepeatedField;

class CredentialOfferBody
{
    private string $url;
    private RepeatedField $credentials;

    /**
     * @param string $url
     * @param RepeatedField $credentials
     */
    public function __construct(string $url, RepeatedField $credentials)
    {
        $this->url = $url;
        $this->credentials = $credentials;
    }

    public static function fromProto(\Bloock\CredentialOfferBody $res): CredentialOfferBody
    {
        return new CredentialOfferBody($res->getUrl(), $res->getCredentials());
    }

    public function toProto(): \Bloock\CredentialOfferBody
    {
        $p = new \Bloock\CredentialOfferBody();
        $p->setUrl($this->url);
        $p->setCredentials($this->credentials);
        return $p;
    }

    /**
     * @return string
     */
    public function getUrl(): string
    {
        return $this->url;
    }

    /**
     * @return RepeatedField
     */
    public function getCredentials(): RepeatedField
    {
        return $this->credentials;
    }
}

File: languages/python/bloock/entity/identity/credential_offer_body.py
from __future__ import annotations

from typing import List

import bloock._bridge.proto.identity_entities_pb2 as proto
from bloock.entity.identity.credential_offer_credential import CredentialOfferCredential


class CredentialOfferBody:
    def __init__(self, url: str, credentials: List[CredentialOfferCredential]) -> None:
        self.url = url
        self.credentials = credentials

    @staticmethod
    def from_proto(c: proto.CredentialOfferBody) -> CredentialOfferBody:
        return CredentialOfferBody(
            url=c.url,
            credentials=list(map(lambda x: CredentialOfferCredential.from_proto(x), c.credentials)),
        )

    def to_proto(self) -> proto.CredentialOfferBody:
        return proto.CredentialOfferBody(
            url=self.url,
            credentials=list(map(lambda x: x.to_proto(), self.credentials)),
        )
